runtime/crab/src/messages/darwinia_messages.rs

```diff
@@ -0,0 +1,277 @@
+	DarwiniaFromThisChainMessageVerifier<WithDarwiniaMessageBridge, R>;
+/// Message payload for Darwinia -> Crab messages.
+pub type FromDarwiniaMessagePayload = FromBridgedChainMessagePayload<WithDarwiniaMessageBridge>;
+/// Encoded Crab Call as it comes from Darwinia.
+pub type FromDarwiniaEncodedCall = FromBridgedChainEncodedMessageCall<crate::Call>;
+/// Messages proof for Darwinia -> Crab messages.
+type FromDarwiniaMessagesProof = FromBridgedChainMessagesProof<Hash>;
+/// Messages delivery proof for Crab -> Darwinia messages.
+type ToDarwiniaMessagesDeliveryProof = FromBridgedChainMessagesDeliveryProof<Hash>;
+/// Call-dispatch based message dispatch for Darwinia -> Crab messages.
+pub type FromDarwiniaMessageDispatch =
+	FromBridgedChainMessageDispatch<WithDarwiniaMessageBridge, Runtime, Ring, S2sBridgeDispatch>;
+
+/// Initial value of `DarwiniaToCrabConversionRate` parameter.
+pub const INITIAL_DARWINIA_TO_CRAB_CONVERSION_RATE: FixedU128 =
+	FixedU128::from_inner(FixedU128::DIV);
+
+frame_support::parameter_types! {
+	/// Darwinia to Crab conversion rate. Initially we treat both tokens as equal.
+	pub storage DarwiniaToCrabConversionRate: FixedU128 = INITIAL_DARWINIA_TO_CRAB_CONVERSION_RATE;
+}
+
+/// Crab -> Darwinia message lane pallet parameters.
+#[derive(Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug)]
+pub enum CrabToDarwiniaMessagesParameter {
+	/// The conversion formula we use is: `CrabTokens = DarwiniaTokens * conversion_rate`.
+	DarwiniaToCrabConversionRate(FixedU128),
+}
+impl MessagesParameter for CrabToDarwiniaMessagesParameter {
+	fn save(&self) {
+		match *self {
+			CrabToDarwiniaMessagesParameter::DarwiniaToCrabConversionRate(ref conversion_rate) => {
+				DarwiniaToCrabConversionRate::set(conversion_rate)
+			}
+		}
+	}
+}
+
+/// Darwinia <-> Crab message bridge.
+#[derive(Clone, Copy, RuntimeDebug)]
+pub struct WithDarwiniaMessageBridge;
+impl MessageBridge for WithDarwiniaMessageBridge {
+	const RELAYER_FEE_PERCENT: u32 = 10;
+	const THIS_CHAIN_ID: ChainId = CRAB_CHAIN_ID;
+	const BRIDGED_CHAIN_ID: ChainId = DARWINIA_CHAIN_ID;
+	const BRIDGED_MESSAGES_PALLET_NAME: &'static str = WITH_CRAB_MESSAGES_PALLET_NAME;
+
+	type ThisChain = Crab;
+	type BridgedChain = Darwinia;
+
+	fn bridged_balance_to_this_balance(bridged_balance: Balance) -> Balance {
+		Balance::try_from(DarwiniaToCrabConversionRate::get().saturating_mul_int(bridged_balance))
+			.unwrap_or(Balance::MAX)
+	}
+}
+
+/// Crab chain from message lane point of view.
+#[derive(Clone, Copy, RuntimeDebug)]
+pub struct Crab;
+impl messages::ChainWithMessages for Crab {
+	type Hash = Hash;
+	type AccountId = AccountId;
+	type Signer = AccountPublic;
+	type Signature = Signature;
+	type Weight = Weight;
+	type Balance = Balance;
+}
+impl messages::ThisChainWithMessages for Crab {
+	type Call = Call;
+
+	fn is_outbound_lane_enabled(lane: &LaneId) -> bool {
+		*lane == [0, 0, 0, 0] || *lane == [0, 0, 0, 1] || *lane == DARWINIA_CRAB_LANE
```

Comment:
```suggestion
		*lane == DARWINIA_CRAB_LANE
```

DARWINIA_CRAB_LANE is expected to be `[0, 0, 0, 0]`